x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/AutomatonQueryOnBinaryDv.java

```diff
@@ -99,12 +96,13 @@ public boolean equals(Object obj) {
             return false;
           }
         AutomatonQueryOnBinaryDv other = (AutomatonQueryOnBinaryDv) obj;
-        return Objects.equals(field, other.field)  && Objects.equals(matchPattern, other.matchPattern);
+        return Objects.equals(field, other.field)  && Objects.equals(matchPattern, other.matchPattern)
+            && Objects.equals(new ByteRunAutomaton(automaton), new ByteRunAutomaton(other.automaton));
```

Comment:
Rebuilding the `ByteRunAutomaton` on equals and hashCode seems overkill. Let's build it once in the ctr, the AutomatonQuery does the same so no reason to differ.